contracts/src/v0.8/ccip/test/pools/GHO/fork/GhoTokenPoolMigrate1_4To1_5/TokenPoolsUpgrade.t.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import {TransparentUpgradeableProxy} from "solidity-utils/contracts/transparent-proxy/TransparentUpgradeableProxy.sol";
+import {ProxyAdmin} from "solidity-utils/contracts/transparent-proxy/ProxyAdmin.sol";
+import {Client} from "../../../../../libraries/Client.sol";
+import {Internal} from "../../../../../libraries/Internal.sol";
+import {ForkBase} from "./ForkBase.t.sol";
+import {UpgradeableLockReleaseTokenPool_Sepolia} from "./LegacyTestnetTokenPools/UpgradeableLockReleaseTokenPool_Sepolia.sol";
+import {UpgradeableBurnMintTokenPool_ArbSepolia} from "./LegacyTestnetTokenPools/UpgradeableBurnMintTokenPool_ArbSepolia.sol";
+
+contract ForkPoolUpgradeAfterMigration is ForkBase {
+  function setUp() public override {
+    super.setUp();
+
+    // #1: deploy new implementation & upgrade token pools
+    vm.selectFork(l1.forkId);
+    _upgradeExistingLockReleaseTokenPool();
+
+    vm.selectFork(l2.forkId);
+    _upgradeExistingBurnMintTokenPool();
+
+    // #2: setProxyPool
+    vm.selectFork(l1.forkId);
+    vm.prank(l1.tokenPool.owner());
+    l1.tokenPool.setProxyPool(l1.proxyPool);
+
+    vm.selectFork(l2.forkId);
+    vm.prank(l2.tokenPool.owner());
+    l2.tokenPool.setProxyPool(l2.proxyPool);
+  }
+
+  function testLockOrBurnViaRouter() public {
+    uint256 amount = 10e18;
+    Client.EVM2AnyMessage memory message = _generateMessage(alice, 1);
+
+    {
+      vm.selectFork(l1.forkId);
+
+      message.tokenAmounts[0] = Client.EVMTokenAmount({token: address(l1.token), amount: amount});
+      uint256 feeTokenAmount = l1.router.getFee(l2.chainSelector, message);
+
+      // router uses 1_5 onRamp
+      assertEq(l1.router.getOnRamp(l2.chainSelector), address(l1.EVM2EVMOnRamp1_5));
+      vm.expectEmit();
+      emit CCIPSendRequested(_messageToEvent(message, l1.EVM2EVMOnRamp1_5, feeTokenAmount, alice, true));
+      vm.prank(alice);
+      l1.router.ccipSend{value: feeTokenAmount}(l2.chainSelector, message);
+    }
+
+    {
+      vm.selectFork(l2.forkId);
+
+      message.tokenAmounts[0] = Client.EVMTokenAmount({token: address(l2.token), amount: amount});
+      uint256 feeTokenAmount = l2.router.getFee(l1.chainSelector, message);
+
+      // router uses 1_5 onRamp
+      assertEq(l2.router.getOnRamp(l1.chainSelector), address(l2.EVM2EVMOnRamp1_5));
+      vm.expectEmit();
+      emit CCIPSendRequested(_messageToEvent(message, l2.EVM2EVMOnRamp1_5, feeTokenAmount, alice, false));
+      vm.prank(alice);
+      l2.router.ccipSend{value: feeTokenAmount}(l1.chainSelector, message);
+    }
+  }
+
+  function testRevertLockOrBurnVia1_2OnRamp() public {
+    uint256 amount = 10e18;
+    {
+      vm.selectFork(l1.forkId);
+      vm.expectRevert(abi.encodeWithSelector(CallerIsNotARampOnRouter.selector, address(l1.EVM2EVMOnRamp1_2)));
```

Comment:
why? I think this shouldn't revert

Reply:
this reverts because you can only have one on ramp per chain, since 1.5 migration deprecates the 1.2 on ramp